src/Workload/Microsoft.NET.Sdk.Maui/Microsoft.NET.Sdk.Maui.csproj

```diff
@@ -131,4 +131,73 @@
+    <_DarcDependency Include="Microsoft.iOS.Sdk" Version="$(MicrosoftiOSSdkPackageVersion)" />
+    <_DarcDependency Include="Microsoft.tvOS.Sdk" Version="$(MicrosofttvOSSdkPackageVersion)" />
+    <_DarcDependency Include="$(MicrosoftNETWorkloadEmscriptenSdkDarcName)" Version="$(MicrosoftNETWorkloadEmscriptenPackageVersion)" />
+    <_DarcDependency Include="Microsoft.AspNetCore.Authorization" Version="$(MicrosoftAspNetCoreAuthorizationPackageVersion)" />
+  </ItemGroup>
+
+  <!-- For each dependency item, execute darc get-asset to list out the version info with feeds -->
+  <Target Name="_DarcGetAssetNugetFeeds" AfterTargets="Build" Outputs="%(_DarcDependency.Identity)">
+
+    <Exec Command="darc get-asset --name %(_DarcDependency.Identity) --version %(_DarcDependency.Version) --max-age 90"
+      ConsoleToMSBuild="True">
+      <Output TaskParameter="ConsoleOutput" ItemName="_DarcGetAssetOutput" />
+    </Exec>
+
+    <!-- Reverse the output since we always want the last feed in the list/output -->
+    <ItemGroup>
+      <_DarcGetAssetOutputReverse
+        Include="@(_DarcGetAssetOutput->Reverse())" />
+    </ItemGroup>
+
+    <!-- Get the first url matching the pattern: ` - https://[feedurlhere]/index.json (NugetFeed)` -->
+    <!-- Using regex for each line of the darc tool output -->
+    <PropertyGroup>
+      <_DarcGetCurrentAssetFeed
+        Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('%(_DarcGetAssetOutputReverse.Identity)', '-\s+http.+/index\.json\s+\(NugetFeed\)'))">$([System.Text.RegularExpressions.Regex]::Match('%(_DarcGetAssetOutputReverse.Identity)', 'http\S+/index\.json'))</_DarcGetCurrentAssetFeed>
+    </PropertyGroup>
+
+    <!-- Finally, add the regex extracted feed to the itemgroup  -->
+    <ItemGroup>
+      <_DarcAssetFeed Include="$(_DarcGetCurrentAssetFeed)" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Takes all the batched up feeds and writes out a nuget.config file -->
+  <Target Name="_DarcAssetFeedCreateNuGetConfig" AfterTargets="_DarcGetAssetNugetFeeds">
+
+    <Message Text="Darc Asset Feeds:" />
+    <Message Text="  %(_DarcAssetFeed.Identity)" />
+
+    <!-- First create an empty nuget config file -->
+    <Exec Command="dotnet new nugetconfig --force --output $(OutputPath)" />
+
+    <!-- Remove the default nuget url since the new template already has it -->
+    <ItemGroup>
+      <_DarcAssetFeed
+        Remove="https://api.nuget.org/v3/index.json"
+        Condition="'%(_DarcAssetFeed.Identity)' == 'https://api.nuget.org/v3/index.json'" />
+    </ItemGroup>
+
+    <!-- Add each of the aggregated feeds to the nuget config -->
+    <Exec Command="dotnet nuget add source %(_DarcAssetFeed.Identity) --configfile $(OutputPath)nuget.config" />
+
+    <!-- Copy to the artifact folders to be archived -->
+    <Copy SourceFiles="$(OutputPath)nuget.config" DestinationFiles="$(PackageOutputPath)/metadata/feeds/$(Version)/$(GitSha)/NuGet.Config" />
```

Comment:
why so many folders you're copying it to?

Reply:
@mattleibow set the original set up - we probably don't need these all, but I was hoping he'd chime in and tell me that too :)

Reply:
for different urls... you may want to use the git sha, branch name or branch version... http://somthing/branch.json or http://somthing/sha.json or http://somthing/version.json